modules/sdk-coin-sol/src/sol.ts

```diff
@@ -428,6 +476,309 @@ export class Sol extends BaseCoin {
+      payload: {
+        id: '1',
+        jsonrpc: '2.0',
+        method: 'getBalance',
+        params: [pubKey],
+      },
+    });
+    if (response.status !== 200) {
+      throw new Error('Account not found');
+    }
+    return response.body.result.value;
+  }
+
+  protected async getAccountInfo(pubKey = ''): Promise<SolDurableNonceFromNode> {
+    const response = await this.getDataFromNode({
+      payload: {
+        id: '1',
+        jsonrpc: '2.0',
+        method: 'getAccountInfo',
+        params: [
+          pubKey,
+          {
+            encoding: 'jsonParsed',
+          },
+        ],
+      },
+    });
+    if (response.status !== 200) {
+      throw new Error('Account not found');
+    }
+    return {
+      authority: response.body.result.value.data.parsed.info.authority,
+      blockhash: response.body.result.value.data.parsed.info.blockhash,
+    };
+  }
+
+  async getTSSSignature(
+    userSigningMaterial: UserSigningMaterial,
+    backupSigningMaterial: BackupSigningMaterial,
+    path = 'm/0',
+    transaction: Transaction
+  ): Promise<Buffer> {
+    const MPC = await Sol.getInitializedMpcInstance();
+
+    const userCombine = MPC.keyCombine(userSigningMaterial.uShare, [
+      userSigningMaterial.bitgoYShare,
+      userSigningMaterial.backupYShare,
+    ]);
+    const backupCombine = MPC.keyCombine(backupSigningMaterial.uShare, [
+      backupSigningMaterial.bitgoYShare,
+      backupSigningMaterial.userYShare,
+    ]);
+
+    const userSubkey = MPC.keyDerive(
+      userSigningMaterial.uShare,
+      [userSigningMaterial.bitgoYShare, userSigningMaterial.backupYShare],
+      path
+    );
+
+    const backupSubkey = MPC.keyCombine(backupSigningMaterial.uShare, [
+      userSubkey.yShares[2],
+      backupSigningMaterial.bitgoYShare,
+    ]);
+
+    const messageBuffer = transaction.signablePayload;
+    const userSignShare = MPC.signShare(messageBuffer, userSubkey.pShare, [userCombine.jShares[2]]);
+    const backupSignShare = MPC.signShare(messageBuffer, backupSubkey.pShare, [backupCombine.jShares[1]]);
+    const userSign = MPC.sign(
+      messageBuffer,
+      userSignShare.xShare,
+      [backupSignShare.rShares[1]],
+      [userSigningMaterial.bitgoYShare]
+    );
+    const backupSign = MPC.sign(
+      messageBuffer,
+      backupSignShare.xShare,
+      [userSignShare.rShares[2]],
+      [backupSigningMaterial.bitgoYShare]
+    );
+    const signature = MPC.signCombine([userSign, backupSign]);
+    const result = MPC.verify(messageBuffer, signature);
+    result.should.equal(true);
+    const rawSignature = Buffer.concat([Buffer.from(signature.R, 'hex'), Buffer.from(signature.sigma, 'hex')]);
+    return rawSignature;
+  }
+
+  /**
+   * Builds a funds recovery transaction without BitGo
+   * @param params
+   */
+  async recover(params: RecoveryOptions): Promise<SolTx> {
+    let isDurableNonceRecovery = false;
+
+    if (_.isUndefined(params.userKey)) {
```

Comment:
Are we able to natively check `!params.userKey` instead of using lodash?

Reply:
great suggestion, i saw that other recover functions were checking this in a similar way, so I followed suit. Alan and I are thinking about doing a bit of refactoring for the code, so this is something that we can definitely change up